action.yml

```diff
@@ -0,0 +1,94 @@
+name: Auto Semver
+description: Auto generate semver
+inputs:
+  major_labels:
+    description: space-separated labels indicate it would bump major version
+    required: false
+    default: major
+  minor_labels:
+    description: space-separated labels indicate it would bump minor version
+    required: false
+    default: minor
+  patch_labels:
+    description: space-separated labels indicate it would bump patch version
+    required: false
+    default: patch
+  token:
+    description: the repo's Github token that use the action
+    required: false
+    default: ${{ github.token }}
+  default_previous_tag_name:
+    description: the default previous tag if there no release tag found
+    required: false
+    default: "0.0.0"
+runs:
+  using: composite
+  steps:
+    - id: last_tag
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ inputs.token }}
+      run: |
+        gh version
+        # prevent there are release title contain space, print count from last column -1
+        # gh release list would show following
+        # <release title> Latest (tag_name) time
+
+        tag_name=$(gh release list -L 1 | awk '{print $(NF - 1)}')
+        [[ $tag_name == '' ]] && tag_name='${{ inputs.default_previous_tag_name }}'
+        echo "::set-output name=tag_name::$tag_name"
+    - id: get_labels
+      shell: bash
+      env:
+        GITHUB_TOKEN: ${{ inputs.token }}
```

Comment:
Add `repository` as input and set here as `env: GH_REPO`. This will allow the action to not need a `actions/checkout`

Reply:
ok

Reply:
fixed